scripts/cargo-install-all.sh

```diff
@@ -134,12 +139,52 @@ for bin in "${BINS[@]}"; do
   binArgs+=(--bin "$bin")
 done
 
+dcouBinArgs=()
+for bin in "${DCOU_BINS[@]}"; do
+  dcouBinArgs+=(--bin "$bin")
+done
+
+source "$SOLANA_ROOT"/scripts/dcou-tainted-packages.sh
+
+excludeArgs=()
+for package in "${dcou_tainted_packages[@]}"; do
+  excludeArgs+=(--exclude "$package")
+done
+
 mkdir -p "$installDir/bin"
 
+# Some binaries (like the notable agave-ledger-tool) need to acitivate
+# the dev-context-only-utils feature flag to build.
+# Build those binaries separately to avoid the unwanted feature unification.
+# Note that `--workspace --exclude <dcou tainted packages>` is needed to really
+# inhibit the feature unification due to a cargo bug. Otherwise, feature
+# unification happens even if cargo build is run only with `--bin` targets
+# which don't depend on dcou as part of dependencies at all.
 (
   set -x
-  # shellcheck disable=SC2086 # Don't want to double quote $rust_version
-  "$cargo" $maybeRustVersion build $buildProfileArg "${binArgs[@]}"
+  # Make sure dcou is really disabled by peeking the (unstable) build plan
+  # output after turning rustc into the nightly mode with RUSTC_BOOTSTRAP=1.
+  # In this way, additional requirement of nightly rustc toolchian is avoided.
+  # Note that `cargo tree` can't be used, because it doesn't support `--bin`.
+  # shellcheck disable=SC2086 # Don't want to double quote $maybeRustVersion
+  if (RUSTC_BOOTSTRAP=1 \
+      "$cargo" $maybeRustVersion build \
+      -Z unstable-options --build-plan \
+      $buildProfileArg "${binArgs[@]}" --workspace "${excludeArgs[@]}" | \
+      grep -q -F '"feature=\"dev-context-only-utils\""'); then
+     echo 'dcou feature activation is incorrctly activated!' && \
+     exit 1
+  fi
+
+  # Build our production binaries without dcou.
+  # shellcheck disable=SC2086 # Don't want to double quote $maybeRustVersion
+  "$cargo" $maybeRustVersion build \
+     $buildProfileArg "${binArgs[@]}" --workspace "${excludeArgs[@]}"
```

Comment:
what's the differences between this one and only --bin pkg...? I assume you use this to avoid getting involved 🤔

Reply:
fyi, `--bin` takes binary name, not crate name.

>  I assume you use this to avoid getting involved 🤔

yeah, manual looping can be another solution (needs bin name => package name translation and `cd` or `cargo-hack). but it's too involved. And build time can take longer.